lib/spack/docs/environments.rst

```diff
@@ -401,6 +401,18 @@ that git clone if ``foo`` is in the environment.
 Further development on ``foo`` can be tested by reinstalling the environment,
 and eventually committed and pushed to the upstream git repo.
 
+If the package being developed supports out-of-source builds then users can use the
```

Comment:
Can this be refactored to be:

1 paragraph explaining the feature

1 paragraph in a `.. warn:` block explaining the limitations.

That way the limitations will be more visible as a warning when the code warning is removed.